lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java

```diff
@@ -0,0 +1,523 @@
+      eventToBeCleaned5.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned6 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId3",
+              TaskHistoryEventType.TERMINATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned6.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned6.setParentBusinessProcessId("sameParentId");
+
+      historyService.create(eventToBeCleaned);
+      historyService.create(eventToBeCleaned2);
+      historyService.create(eventToBeCleaned3);
+      historyService.create(eventToBeCleaned4);
+      historyService.create(eventToBeCleaned5);
+      historyService.create(eventToBeCleaned6);
+
+      HistoryCleanupJob job = new HistoryCleanupJob(taskanaEngine, null, null);
+      job.run();
+
+      assertThat(historyService.createTaskHistoryQuery().count()).isZero();
+    }
+
+    @Test
+    @WithAccessId(user = "admin")
+    void should_NotCleanHistoryEvents_When_SameParentBusinessTrueAndActiveTaskInParentBusiness()
+        throws Exception {
+
+      TaskHistoryEvent eventToBeCleaned =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId1",
+              TaskHistoryEventType.CREATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned2 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId1",
+              TaskHistoryEventType.COMPLETED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned2.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned2.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned3 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId2",
+              TaskHistoryEventType.CREATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned3.setCreated(Instant.now().minus(1, ChronoUnit.DAYS));
+      eventToBeCleaned3.setParentBusinessProcessId("sameParentId");
+
+      historyService.create(eventToBeCleaned);
+      historyService.create(eventToBeCleaned2);
+      historyService.create(eventToBeCleaned3);
+
+      HistoryCleanupJob job = new HistoryCleanupJob(taskanaEngine, null, null);
+      job.run();
+
+      assertThat(historyService.createTaskHistoryQuery().count()).isEqualTo(3);
+    }
+
+    @Test
+    @WithAccessId(user = "admin")
+    void
+        should_NotCleanHistoryEvents_When_MinimumAgeOfOtherEndStateEventInParentBusinessNotReached()
+            throws Exception {
+      TaskHistoryEvent eventToBeCleaned =
```

Comment:
Do you think 3 events would be enough?

Reply:
I do not think it is possible, because the test tests if the events of another task with same parent ID is not qualified to be deleted only because of the minimum age, then all the events should not be deleted.  

The test requires minimum 2 tasks, one with qualified events and one with unqualified events. Each task require two events, CREATE and an endstate event